Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/UnityProjectInfo.cs

```diff
@@ -259,6 +269,14 @@ private void CorrectReferences(Dictionary<string, AssemblyDefinitionInfo> asmDef
                         }
                     }
                 }
+
+                if (ImpledDefinesForAsmDefs.TryGetValue(asmDefPair.Key, out List<string> defines))
+                {
+                    foreach (string define in defines)
+                    {
+                        asmDefPair.Value.DefineConstraints.Add(define);
+                    }
+                }
```

Comment:
nit: It seems like this should be broken out into a CorrectConstantDefines function

Reply:
I will rename the entire function, I just think we should loop over asmdefs once.

---
In reply to: [366027569](https://github.com/microsoft/MSBuildForUnity/pull/104#discussion_r366027569) [](ancestors = 366027569)